gmso/formats/mol2.py

```diff
@@ -43,36 +43,63 @@ def from_mol2(filename, site_type="atom"):
         >>> top = Topology.load('myfile.mol2')
         >>> mbuild_compound = to_mbuild(top)
     """
-    msg = "Provided path to file that does not exist"
     path = Path(filename)
     if not path.exists():
+        msg = "Provided path to file that does not exist"
         raise OSError(msg)
     # Initialize topology
     topology = Topology(name=path.stem)
     # save the name from the filename
-    f = open(path, "r")
-    line = f.readline()
-    while f:
-        # check for header character in line
-        if line.startswith("@<TRIPOS>"):
-            # if header character in line, send to a function that will direct it properly
-            line = parse_record_type_indicator(f, line, topology, site_type)
-        elif line == "":
-            break
-        else:
-            # else, skip to next line
-            line = f.readline()
-    f.close()
+    with open(path, "r") as f:
+        line = f.readline()
+        while f:
+            # check for header character in line
+            if line.strip().startswith("@<TRIPOS>"):
+                # if header character in line, send to a function that will direct it properly
+                line = _parse_record_type_indicator(
+                    f, line, topology, site_type
+                )
+            elif line == "":
+                # check for the end of file
+                break
+            else:
+                # else, skip to next line
+                line = f.readline()
     topology.update_topology()
```

Comment:
I tried my best to write something that could use a for loop instead of a while loop. It starts to get tricky again when it finds a header in the line. That passes it to the `_parse_record_type_indicator` function, which reads through the rest of that heading. In the case of a record type indicator (rti) starting a new section without a space after the previous section, that line must be passed back to the parser in order to continue forwards and identify which rti you are dealing with. With a for loop, every time through you must read the next line. In the instance as discussed above:
"""
@<TRIPOS>ATOM
Atom information
@<TRIPOS>BOND
bond information
"""

The next line cannot be read automatically because it will already be read when looking to find the end of the ATOM rti section. 

Does that make sense? There may be a way to accomplish this while still using a for loops, but I tried exactly what you suggested and I run into this error still.